### DISCUSSION TITLE
Possible to get spatial object from duckdb query in Python ?

### DISCUSSION THREAD

**Discussion by glaroc** [duckdb/duckdb#11534] [Q&A] [answered]
Hi, is it possible to get a Geopandas spatial object or a geojson string from a duckdb.sql() query in Python ? 

I have a spatial query that looks like this

````{python}
 res = duckdb.sql("COPY (SELECT wdpa.* FROM ST_Read('/data/WDPA_protected_areas.gpkg') as wdpa, ST_Read('/data/natural_earth_adm_1.fgb') as country WHERE country.NAME_EN='Canada' AND ST_Intersects(country.geom,wdpa.geom)) TO 'canada_wdpa.geojson' WITH (FORMAT GDAL, DRIVER 'GeoJSON');")
````

However, I would like to have the geojson string returned to `res` instead of saving to a file. If I could return the SELECT query with  `duckdb.sql().df()` as a geopandas dataframe, I could then convert it into a geojson string, but I don't know whether this is possible. I've also tried to send to '/dev/stdout' or to '/vsistdout/', but this won't transfer the result to Python. 

Any ideas ?

**Comment by Tishj** [comment, marked as answer]
This currently isn't possible

The easiest solution to this would be to bundle the spatial extension so it's always available.
However this is not really something we're very keen on doing because of the size of the spatial extension.

With that in mind, we'd need to look into ways to either allow the Spatial extension to extend the python client or detect that Spatial datatypes are used and add in special conversion logic for that case.
